File: src/main/java/com/mtsahakis/quiz/Factorial.java
package com.mtsahakis.quiz;

public class Factorial {

    public static int iterativeFactorialFor(int num) {
        if (num < 0) {
            throw new IllegalArgumentException("factorial input cannot be negative");
        }

        int result = 1;

        for (int i = 1; i <= num; i++) {
            result *= i;
        }

        return result;
    }

    public static int iterativeFactorialWhile(int num) {
        if (num < 0) {
            throw new IllegalArgumentException("factorial input cannot be negative");
        }

        int result = 1;
        int i = num;

        while (i > 1) {
            result *= i;
            i--;
        }

        return result;
    }

    public static int recursiveFactorial(int num) {
        if (num < 0) {
            throw new IllegalArgumentException("factorial input cannot be negative");
        }

        if (num == 0) {
            return 1;
        }

        return num * recursiveFactorial(num - 1);
    }

}

File: src/main/kotlin/com/mtsahakis/quiz/Factorial.kt
package com.mtsahakis.quiz

fun iterativeFactorialFor(num: Int): Int {
    if (num < 0) {
        throw IllegalArgumentException("factorial input cannot be negative")
    }

    var result = 1

    for (i in 1..num) {
        result *= i
    }

    return result
}

fun iterativeFactorialWhile(num: Int): Int {
    if (num < 0) {
        throw java.lang.IllegalArgumentException("factorial input cannot be negative")
    }

    var result = 1
    var i = num

    while (i > 1) {
        result *= i
        i--
    }

    return result
}

fun recursiveFactorial(num: Int): Int {
    if (num < 0) {
        throw java.lang.IllegalArgumentException("factorial input cannot be negative")
    }

    if (num == 0) {
        return 1
    }

    return num * recursiveFactorial(num - 1)
}